server/client/client.go

```diff
@@ -0,0 +1,187 @@
+package client
+
+import (
+	"context"
+	"github.com/digitalocean/godo"
+	"golang.org/x/oauth2"
+)
+
+//go:generate mockgen -destination=../mocks/mock_client.go -package=mocks github.com/phillipahereza/mattermost-plugin-digitalocean/server/client DigitalOceanService
+
+// DigitalOceanClient manages communication with DigitalOcean V2 API.
+type DigitalOceanClient struct {
+	Client *godo.Client
+}
+
+// K8sNode represents a node in a Kubernetes cluster with the node pool name
+type K8sNode struct {
+	*godo.KubernetesNode
+	NodePoolName string
+}
+
+// TokenSource contains the access token used for authentication
+type TokenSource struct {
+	AccessToken string
+}
+
+// Token returns the access token
+func (t *TokenSource) Token() (*oauth2.Token, error) {
+	token := &oauth2.Token{
+		AccessToken: t.AccessToken,
+	}
+	return token, nil
+}
+
+// DigitalOceanService is an interface for interfacing with the DigitalOcean API endpoints
+type DigitalOceanService interface {
```

Comment:
Maybe it's better to name this Client to be a little bit more consistent with ` DigitalOceanClient` on line 12

Reply:
> Maybe it's better to name this Client to be a little bit more consistent with ` DigitalOceanClient` on line 12

Rename to client

Reply:
Won't the name collide with DigitalOceanClient of type struct on line 12

Reply:
I doubt. As we have no other types named Client